internal/controller/vrg_volsync.go

```diff
@@ -29,8 +34,23 @@ func (v *VRGInstance) restorePVsAndPVCsForVolSync() (int, error) {
 
 	for _, rdSpec := range v.instance.Spec.VolSync.RDSpec {
 		failoverAction := v.instance.Spec.Action == ramendrv1alpha1.VRGActionFailover
-		// Create a PVC from snapshot or for direct copy
-		err := v.volSyncHandler.EnsurePVCfromRD(rdSpec, failoverAction)
+
+		var err error
+
+		cg, ok := rdSpec.ProtectedPVC.Labels[ConsistencyGroupLabel]
```

Comment:
(nit) This line can be moved inside the `if` block

Reply:
will change it in another pr once https://github.com/RamenDR/ramen/pull/1472 merged.